local-tests/setup/tinny-environment.ts

```diff
@@ -0,0 +1,366 @@
+   * This setup differentiates between local and production environments, adjusts node attestation checks,
+   * and sets network-specific parameters. The function ensures the client is connected and ready before proceeding.
+   *
+   * The LitNodeClient is configured differently based on the network:
+   * - LOCALCHAIN: Uses custom settings for local testing, with node attestation disabled.
+   * - MANZANO (or other specified testnets): Configures for specific network environments with node attestation enabled.
+   *
+   * Logs the process and exits if the client is not ready after attempting to connect.
+   */
+
+  async setupLitNodeClient() {
+    console.log('[𐬺🧪 Tinny Environment𐬺] Setting up LitNodeClient');
+
+    if (this.network === LIT_TESTNET.LOCALCHAIN) {
+      this.litNodeClient = new LitNodeClient({
+        litNetwork: 'custom',
+        bootstrapUrls: this.processEnvs.BOOTSTRAP_URLS,
+        rpcUrl: this.processEnvs.LIT_RPC_URL,
+        debug: this.processEnvs.DEBUG,
+        checkNodeAttestation: false, // disable node attestation check for local testing
+        contractContext: networkContext as LitContractContext,
+      });
+    } else if (this.network === LIT_TESTNET.MANZANO) {
+      this.litNodeClient = new LitNodeClient({
+        litNetwork: this.network, // 'habanero' or 'manzano'
+        checkNodeAttestation: true,
+        debug: this.processEnvs.DEBUG,
+      });
+    } else {
+      this.litNodeClient = new LitNodeClient({
+        litNetwork: this.network,
+        checkNodeAttestation: false,
+        debug: this.processEnvs.DEBUG,
+      });
+    }
+
+    await this.litNodeClient.connect();
+
+    if (!this.litNodeClient.ready) {
+      console.error('❌ litNodeClient not ready');
+      process.exit();
+    }
+  }
+
+  /**
+   * Retrieves the environment configuration.
+   * @returns The TinnyEnvConfig object containing the environment configuration.
+   */
+  getEnvConfig(): TinnyEnvConfig {
+    return {
+      rpc: this.rpc,
+      litNodeClient: this.litNodeClient,
+      network: this.network,
+      processEnvs: this.processEnvs,
+    };
+  }
+
+  /**
+   * Creates a new person with the given name.
+   * @param name - The name of the person.
+   * @returns The newly created person.
+   * @throws Error if the name is not provided.
+   */
+  async createNewPerson(name: string) {
+    console.log('[𐬺🧪 Tinny Environment𐬺] Creating new person:', name);
+    if (!name) {
+      throw new Error('Name is required');
+    }
+    const key = await this.getAvailablePrivateKey();
+    const privateKey = key.privateKey;
+    const envConfig = this.getEnvConfig();
+
+    console.log(
+      '[𐬺🧪 Tinny Environment𐬺] Creating new person with private key:',
+      JSON.stringify(key)
+    );
+    console.log(
+      '[𐬺🧪 Tinny Environment𐬺] Creating new person with envConfig:',
+      envConfig
+    );
+
+    const person = new TinnyPerson({
+      privateKey,
+      envConfig,
+    });
+
+    await person.spawn();
```

Comment:
1. Don't we need to 'release' the private key now that we're done?
2. Aren't tests going to use this `person`'s private key _internally_ as they sign things?  It seems like although a particular key may have been available when _this method was called_, it may or may not be available _later on_ when the tests are actually running, and there could still be nonce issues?